jekyll/_cci2/build-publish-snap-packages.md

```diff
@@ -0,0 +1,176 @@
+
+```
+...
+    steps:
+      - checkout
+      - run:
+          name: "Build Snap"
+          command: snapcraft
+...
+```
+
+On CircleCI, this single command is needed to actually build your snap. This will run Snapcraft, which will then go through all of it's build steps and generate a `.snap` file for you. This file will typically be in the format of `<snap-name>-<snap-version>-<system-arch>.snap`.
+
+## Testing
+
+Unit testing your software's code has been covered extensively in [our blog][blog], [our docs][docs], and around the Internet. Searching for your language/framework and unit testing or CI will turn up tons of information.
+
+You'll likely want to create a `job` before building the snap that pull project dependencies, any pre-checks you'd want to do, testing, and compiling.
+
+Building snaps on CircleCI means we end with a `.snap` file which we can test in addition to the code that created it. How you test the snap itself is up to you. Some users will attempt to install the snap in various distros and then run a command to make sure that installation process works. Snapcraft offers a build fleet for spreadtesting that allows you to test snaps on different distros, after you've already tested the code itself. This can be found [here][snapcraft-build].
+
+## Publishing
+
+Publishing a snap is more or less a two-step process. The first step which is done once, is to create a Snapcraft "login file" on your local machine that we upload to CircleCI. Assuming your local machine already has these tools installed and you are logged in to the Snapcraft Store (`snapcraft login`), we use the command `snapcraft export-login snapcraft.login` to generate a login file called `snapcraft.login`. As we don't want this file visible to the public or stored in the Git repository, we will base64 encode this file and store it in a [private environment variable][private-envars] called `$SNAPCRAFT_LOGIN_FILE`.
+
+Here's on this might look on a Linux machine:
+
+```
+snapcraft login
+# follow prompts for logging in with an Ubuntu One account
+snapcraft export-login snapcraft.login
+base64 snapcraft.login | xsel --clipboard
+```
+
+Once the base64 encoded version of the file is stored on CircleCI as a private environment variable, we can then use it within a build to automatically publish to the store.
+
+```
+...
+      - run:
+          name: "Publish to Store"
+          command: |
+            mkdir .snapcraft
+            echo $SNAPCRAFT_LOGIN_FILE | base64 --decode --ignore-garbage > .snapcraft/snapcraft.cfg
+            snapcraft push *.snap --release stable
+...
+```
+
+In this example, Snapcraft automatically looks for login credentials in `.snapcraft/snapcraft.cfg` so we decode the environment variable we made previously right into that location. `snapcraft push` is then used to upload the .snap file into the Snap Store.
+
+### Uploading vs Releasing
```

Comment:
This section feels a little out of sync with the configuration. It's referencing commands from above, but the pattern so far has been to (1) describe action (2) perform action. This should match that pattern, or (my preference) be called out in the config itself.